src/path.cr

```diff
@@ -545,16 +545,18 @@ struct Path
   # unless *base* is given, in which case it will be used as the reference path.
   #
   # ```
-  # Path["foo"].expand             # => Path["/current/path/foo"]
-  # Path["~/crystal/foo"].expand   # => Path["/home/crystal/foo"]
-  # Path["baz"].expand("/foo/bar") # => Path["/foo/bar/baz"]
+  # Path["foo"].expand                 # => Path["/current/path/foo"]
+  # Path["~/foo"].expand(home: true)   # => Path["/home/crystal/foo"]
+  # Path["~/foo"].expand(home: "/bar") # => Path["/bar/foo"]
+  # Path["baz"].expand("/foo/bar")     # => Path["/foo/bar/baz"]
   # ```
   #
-  # *home* specifies the home directory which `~` will expand to.
+  # "~/" is expanded to the value passed to *home* or the user's home directory
+  # when *home* is `true`
   # If *expand_base* is `true`, *base* itself will be exanded in `Dir.current`
   # if it is not an absolute path. This guarantees the method returns an absolute
   # path (assuming that `Dir.current` is absolute).
-  def expand(base : Path | String = Dir.current, *, home = Path.home, expand_base = true) : Path
+  def expand(base : Path | String = Dir.current, *, home = nil, expand_base = true) : Path
```

Comment:
I'd suggest to only accept `Path | String | Nil` for `home`.
Opinions?

Reply:
Why not allowing `Bool` as a shorthand as well?

Reply:
`true` is the most common and likely use and allows for future `~username/foo` expansion.

Reply:
I'd question whether we even want/need `~username` expansion at all. But that is a different discussion. As is how this would be integrated into the API. We don't need to consider that *right now*. Expanding the type to accept `Bool` later is easy and doesn't break anything. We should only consider if `Bool` makes sense at the present.

@Sija `Path.home` is more explict, see https://github.com/crystal-lang/crystal/pull/7903#issuecomment-515983270 
Also, it avoids doubts about the difference between `home: nil` and `home: false`.

Reply:
To remove of the `Bool` argument would anyone be willing to look at the spec failures after adding ```home = Path.home if home == true``` to `src/spec/std_path.cr`?

Reply:
@didactic-drunk Which failure? There's at least one `home: true` in the code.

Also, `it_expands_path` should call `it "..." do ... end`, otherwise the "tests" don't run inside an "it" block and they are not properly reported. Same goes for `it_normalizes_path`.

Reply:
Though I guess there's some old code there that's incorrect and wasn't written by you? I'm not sure...

Reply:
It's also easier if you add a type restriction to the `home` arguments with the types you want to allow, then you'll get a compile error where you are passing a `Bool`.

Reply:
> Which failure? There's at least one `home: true` in the code.

Line 22 `src/std/path_spec.cr` .

Reply:
I plan on adding type restrictions after resolving this issue.

Reply:
> @didactic-drunk Which failure? There's at least one `home: true` in the code.
> 
> Also, `it_expands_path` should call `it "..." do ... end`, otherwise the "tests" don't run inside an "it" block and they are not properly reported. Same goes for `it_normalizes_path`.

Most of the path specs weren't written by me.

Are `it` blocks in `assert_paths_raw` sufficient?

Reply:
@asterite The `it` spec blocks are in `assert_paths_raw`. That method is probably misnamed, but everything works as expected.

Reply:
When line 22 is uncommented all home directory specs fail. I don't understand why.  The output shows bizarre half windows paths:
```
       Expected: Path.windows("C:\\Users\\Crystal\\a")
            got: Path.windows("\\home\\crystal\\a")
```